cloudinit/sources/__init__.py

```diff
@@ -317,18 +317,18 @@ def override_ds_detect(self):
         - TODO: commandline argument is used (ci.ds=OpenStack)
         """
         return self.sys_cfg.get("datasource_list", []) in (
-            [self.dslist],
-            [self.dslist, "None"],
+            [self.dsname],
```

Comment:
What is None comes first?

Would something like this be better?
```python
ds_list = self.sys_cfg.get("datasource_list", [])
return self.dsname in ds_list and any(
    [
        len(ds_list) == 1,
        "None" in ds_list and len(ds_list) == 2,
    ]
)
```

Reply:
> What is None comes first?

If that can happen, this improvement looks good to me.

Reply:
since it's user-modifyable, it's unlikely, but it can.

Reply:
I mean...we specifically document a datasource alongside none, so I don't think it was an unreasonable thought. However, I wasn't thinking that the datasource list is ordered, so `None` coming at front doesn't actually work. So....nevermind.